### PR TITLE
Show extras and dev dependencies in `uv tree`

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

Closes https://github.com/astral-sh/uv/issues/5759.

### CODE REVIEW

File: crates/uv/tests/tree.rs [hunk 1]
```diff
@@ -386,3 +432,115 @@ fn dev_dependencies() -> Result<()> {
...
+        dev = ["anyio"]
+    "#,
+    )?;
+
+    uv_snapshot!(context.filters(), context.tree(), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    project v0.1.0
+    ├── flask[dotenv] v3.0.2
```
Comment (charliermarsh) [inline crates/uv/tests/tree.rs:460]: This means: `flask[dotenv]` was a dependency.

File: crates/uv/tests/tree.rs [hunk 2]
```diff
@@ -386,3 +432,115 @@ fn dev_dependencies() -> Result<()> {
...
+    │   ├── blinker v1.7.0
+    │   ├── click v8.1.7
+    │   │   └── colorama v0.4.6
+    │   ├── itsdangerous v2.1.2
+    │   ├── jinja2 v3.1.3
+    │   │   └── markupsafe v2.1.5
+    │   ├── werkzeug v3.0.1
+    │   │   └── markupsafe v2.1.5
+    │   └── python-dotenv v1.0.1 (extra: dotenv)
```
Comment (charliermarsh) [inline crates/uv/tests/tree.rs:469]: This means: `python-dotenv` was included due to the `dotenv` extra.

File: crates/uv/tests/tree.rs [hunk 3]
```diff
@@ -386,3 +432,115 @@ fn dev_dependencies() -> Result<()> {
...
+    "#,
+    )?;
+
+    uv_snapshot!(context.filters(), context.tree().arg("--invert"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    blinker v1.7.0
+    └── flask v3.0.2
+        └── project[dotenv] v0.1.0
```
Comment (charliermarsh) [inline crates/uv/tests/tree.rs:513]: The inverted form is kind of bad, the annotations are sort of on the wrong node, but it's honestly really hard to improve. I considered omitting them altogether.

File: crates/uv-resolver/src/lock.rs [hunk 4]
```diff
@@ -2659,8 +2663,12 @@ impl<'env> TreeDisplay<'env> {
         invert: bool,
     ) -> Self {
         let mut non_roots = FxHashSet::default();
-        let mut edges: FxHashMap<_, Vec<_>> =
-            FxHashMap::with_capacity_and_hasher(lock.by_id.len(), FxBuildHasher);
+
+        // Index all the dependencies. We could read these from the `Lock` directly, but we have to
+        // support `--invert`, so we might as well build them up in either case.
```
Comment (charliermarsh) [inline crates/uv-resolver/src/lock.rs:2668]: Honestly wondering if `--invert` should have a totally separate impl.

**Review by konstin — APPROVED** [review]

**Pull request merged**